services/storage/src/simcore_service_storage/modules/celery/backends/_redis.py

```diff
@@ -1,21 +1,44 @@
+import logging
+from datetime import timedelta
 from typing import Final
 
+from celery.result import AsyncResult
 from servicelib.redis._client import RedisClientSDK
 
-from ..models import TaskContext, TaskData, TaskID, TaskUUID, build_task_id_prefix
+from ..models import TaskContext, TaskID, TaskMetadata, TaskUUID, build_task_id_prefix
 
 _CELERY_TASK_META_PREFIX: Final[str] = "celery-task-meta-"
+_CELERY_TASK_METADATA_PREFIX: Final[str] = "celery-task-metadata-"
+_CELERY_TASK_ID_KEY_ENCODING = "utf-8"
 _CELERY_TASK_ID_KEY_SEPARATOR: Final[str] = ":"
 _CELERY_TASK_SCAN_COUNT_PER_BATCH: Final[int] = 10000
-_CELERY_TASK_ID_KEY_ENCODING = "utf-8"
+
+_logger = logging.getLogger(__name__)
 
 
-class RedisTaskStore:
+def _build_key(task_id: TaskID) -> str:
+    return _CELERY_TASK_META_PREFIX + task_id
+
+
+class RedisTaskMetadataStore:
     def __init__(self, redis_client_sdk: RedisClientSDK) -> None:
         self._redis_client_sdk = redis_client_sdk
 
-    async def get_task_uuids(self, task_context: TaskContext) -> set[TaskUUID]:
-        search_key = build_task_id_prefix(task_context) + _CELERY_TASK_ID_KEY_SEPARATOR
+    async def exists(self, task_id: TaskID) -> bool:
+        n = await self._redis_client_sdk.redis.exists(_build_key(task_id))
+        assert isinstance(n, int)  # nosec
+        return n > 0
+
+    async def get(self, task_id: TaskID) -> TaskMetadata | None:
+        result = await self._redis_client_sdk.redis.get(_build_key(task_id))
+        return TaskMetadata.model_validate_json(result) if result else None
+
+    async def get_uuids(self, task_context: TaskContext) -> set[TaskUUID]:
```

Comment:
```suggestion
    async def list_uuids(self, task_context: TaskContext) -> set[TaskUUID]:
```

Reply:
What is the difference between TaskID and TaskUUID? when do you need one or the other?

Reply:
`TaskUUID` is a UUID. `TaskID` is composed by `TaskContext` (values serialized and separated by :) + `TaskUUID`.

e.g. TaskUUID = d439a420-daad-4446-a297-924367df2940, TaskID = osparc:42:d439a420-daad-4446-a297-924367df2940